src/Command/CommandBase.php

```diff
@@ -1127,36 +1128,36 @@ public static function getAliasCache() {
    * @return mixed
    * @throws AcquiaCliException
    */
-  protected function doGetApplicationFromAlias($application_alias) {
+  protected function doGetApplicationFromAlias($application_alias): mixed {
+    if (!strpos($application_alias, ':')) {
+      $application_alias = '*:' . $application_alias;
+    }
     $acquia_cloud_client = $this->cloudApiClientService->getClient();
-    $acquia_cloud_client->addQuery('filter', 'hosting=@*:' . $application_alias);
+    $acquia_cloud_client->addQuery('filter', 'hosting=@' . $application_alias);
     // Allow Cloud users with 'support' role to resolve aliases for applications to
     // which they don't explicitly belong.
     $account_resource = new Account($acquia_cloud_client);
     $account = $account_resource->get();
     if ($account->flags->support) {
       $acquia_cloud_client->addQuery('all', 'true');
     }
-    $customer_applications = $acquia_cloud_client->request('get', '/applications');
-    $site_prefix = '';
-    if ($customer_applications) {
-      $customer_application = $customer_applications[0];
-      $site_id = $customer_application->hosting->id;
-      $parts = explode(':', $site_id);
-      $site_prefix = $parts[1];
-    }
-    else {
+    $applications_resource = new Applications($acquia_cloud_client);
+    $customer_applications = $applications_resource->getAll();
+    if (count($customer_applications) === 0) {
       throw new AcquiaCliException("No applications found");
     }
-
-    if ($site_prefix !== $application_alias) {
-      throw new AcquiaCliException("Application not found matching the alias {alias}", ['alias' => $application_alias]);
+    if (count($customer_applications) > 1) {
+      $callback = static function ($element) {
+        return $element->hosting->id;
+      };
+      $aliases = array_map($callback, (array) $customer_applications);
+      $this->io->error(sprintf("Multiple applications match this alias. Use one of the following aliases instead: %s", implode(', ', $aliases)));
+      throw new AcquiaCliException(sprintf("An application matching the alias %s was found in multiple hosting realms.", $application_alias));
     }
 
-    $this->logger->debug("Found application {$customer_application->uuid} matching alias $application_alias.");
+    $customer_application = $customer_applications[0];
 
-    // Remove the host=@*.$drush_site query as it would persist for future requests.
-    $acquia_cloud_client->clearQuery();
```

Comment:
Not true, the client is generated uniquely by a factory on each call to `$this->cloudApiClientService->getClient();`